### DISCUSSION TITLE
[Bug] Error when creating custom domain with OVH

### DISCUSSION THREAD

**Discussion by caouic** [webstudio-is/webstudio-community#131] [Q&A]
Hi, when I create a custom domain. The CNAME entries are OK. But TXT entries seems having two '' at the beginning of the name entry.
OVH refuse to create a TXT entry with two '_'
With one '-' the entry is created
![Capture d'écran 2024-03-22 111120](https://github.com/webstudio-is/webstudio-community/assets/29862059/f482ff9a-093d-482a-8caa-f2bf05ea4c2b)
![Capture d'écran 2024-03-22 112646](https://github.com/webstudio-is/webstudio-community/assets/29862059/9cccd679-a39b-4683-91ae-6dea4ac1aacd)
![Capture d'écran 2024-03-22 112925](https://github.com/webstudio-is/webstudio-community/assets/29862059/410be767-e566-4e4b-9e62-0634a167f47e)

**Comment by kof** [comment]
I don't really understand the problem, did you compare the txt record? is it having exactly what webstudio gives you? did you use copy button? or copied manually and made a mistake?

**Reply by caouic** [reply]
Hi, Webstudio gives me __webstudio_is but OVH doesn't allow it. _webstudio_is  is the only format allowed by OVH for TXT entry.

**Reply by caouic** [reply]
I use copy button. I don't enter them manually.

**Reply by kof** [reply]
Oh so OVH doesn't allow to enter a correct TXT string? wow. This is highly unusual, I don't think there is a standard that dictates to do that ... something must be super bad at OVH, I would ask them for the reason they don't allow that TXT string

**Reply by kof** [reply]
>  All printable ASCII characters are permitted in the attribute value.

https://www.rfc-editor.org/rfc/rfc1464.txt

I strongly recommend to stop using that registrar if possible or alternatively use cloudflare on top of it (doesn't require migrating) https://docs.webstudio.is/university/foundations/publishing-and-custom-domains#how-to-add-a-custom-domain-on-registrars-that-dont-allow-root-cnames-records

**Reply by caouic** [reply]
Sorry to insist, but other customers using OVH will have the same issue. 
The entry you provide for TXT entry with __domain_com is not standard. 

Here is a answer « The syntax `__domain_com` for a TXT type DNS entry is not valid according to usual standards. TXT records must be associated with a specific domain or subdomain and should not contain invalid characters or incorrect formats. The presence of two consecutive underscores (`__`) at the beginning of the entry does not comply with DNS record conventions.

The correct entry that you mentioned, `_domain_com`, is more in line with standards. TXT records can use a single underscore to start a record name, which is often used for technical specifications like SPF or DKIM records. For example, a DKIM record might look like `selector._domainkey.yourdomain.com` where `selector` is the DKIM selector and `yourdomain.com` is your domain.

If you are encountering an error with a hosting provider, it is possible that the error is due to incorrect syntax or specific restrictions of the hosting provider regarding the allowed formats for TXT records. It is recommended to consult the hosting provider's documentation or contact their technical support for precise instructions on how to format TXT records correctly for their system. »

**Reply by kof** [reply]
Did they tell you "two consecutive underscores" are not allowed? I don't see it in a the standards

**Reply by kof** [reply]
To me it seems like they are in violation with standards and came up with random rules like "two conscutive underscores are not allowed but one underscore is ok" this smells badly :)

**Reply by caouic** [reply]
I understand. I want just to report you a problem that could impact future customers. All TXT entries I have to set in the past are this format _domain_com

**Reply by caouic** [reply]
For the moment, I am stuck with this and all my domain are on OVH. I can’t move them to another hosting provider.

**Reply by kof** [reply]
A quick solution would be to connect using cloudflare, this doesn't require switching the dns provider, this video should help https://docs.webstudio.is/errors/connecting-your-hostinger-domain-to-webstudio

**Reply by kof** [reply]
If you are in contact with OVH customer support do tell them they are wrong about this error and they need to fix it.

Send them link to the spec
https://www.rfc-editor.org/rfc/rfc1464.txt

**Comment by kof** [comment]
follow this issue for updates https://github.com/webstudio-is/webstudio/issues/3124
